### PR TITLE
Feat/4 rdf graph loader

### PR DESCRIPTION
**Pull request by msesboue**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -150,4 +150,6 @@ dmypy.json
 .pytype/
 
 # Cython debug symbols
-cython_debug/
\ No newline at end of file
+cython_debug/
+buzz_el/rdf_kg_2_el_loader.py
```
Comment (schmarion) [inline .gitignore:154]: does it make sense to keep this here ?

File: buzz_el/commons/utils.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    Parameters
+    ----------
+    uri : str
+        The string to test.
+
+    Returns
+    -------
+    bool
+        Whether the string is da URL.
```
Comment (schmarion) [inline buzz_el/commons/utils.py:15]: is a  instead of da ?

**Review by schmarion — COMMENTED** [review]
Can you update the requirements  file ?

File: .gitignore (hunk 1 above)
Comment (msesboue) [reply .gitignore:154]: The cython debug I would say yes. It comes from a standard python project gitignore.

File: buzz_el/graph_loader/graph_loader.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+from buzz_el.commons.utils import is_valid_url
+
+
+class GraphLoader:
+    """
+    A class for loading a knowledge graph, extracting entity labels, and building an index.
+
+    Parameters
+    ----------
+    kg_file_path : Any
```
[the remark this answers is not in the document]
Comment (msesboue) [reply buzz_el/graph_loader/graph_loader.py:16]: I made it an os.PathLike

File: buzz_el/graph_loader/graph_loader.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        self.ent_str_sparql_q = sparql_q_ent_labels
+
+    def load_kg_from_file(self) -> None:
+        """Load the knowledge graph from the specified file."""
+        self.kg = self.kg.parse(self._kg_file_path)
+
+    def build_entity_strings_index(self) -> None:
+        """Build the index of entity strings by querying the knowledge graph."""
+        self._entity_str_idx = defaultdict(set)
+        for r in self.kg.query(self.ent_str_sparql_q):
```
[the remark this answers is not in the document]
Comment (msesboue) [reply buzz_el/graph_loader/graph_loader.py:121]: I left the getter as is but put ran the method in the ini;

File: .gitignore (hunk 1 above)
Comment (schmarion) [reply .gitignore]: Sorry, my comment was on line 154 `buzz_el/rdf_kg_2_el_loader.py`, this is for your local right ?

Comment (msesboue) [reply .gitignore]: Yes it wa my local file, but I removed it. Didn't I ?

File: buzz_el/graph_loader/graph_loader.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (schmarion) [reply buzz_el/graph_loader/graph_loader.py:17]: Just type change missing line 17

**Pull request merged**